plugins/pantheon-files-ctags/plugin.vala

```diff
@@ -330,20 +330,300 @@ public class Marlin.Plugins.CTags : Marlin.Plugins.Base {
+
+            var color_button_green = new Gtk.Button ();
+            color_button_green.halign = Gtk.Align.CENTER;
+            color_button_green.height_request = BUTTON_WIDTH;
+            color_button_green.width_request = BUTTON_HEIGHT;
+            var color_button_green_context = color_button_green.get_style_context ();
+            color_button_green_context.add_class ("color-button");
+            color_button_green_context.add_class ("color-green");
+
+            var color_button_blue = new Gtk.Button ();
+            color_button_blue.halign = Gtk.Align.CENTER;
+            color_button_blue.height_request = BUTTON_WIDTH;
+            color_button_blue.width_request = BUTTON_HEIGHT;
+            var color_button_blue_context = color_button_blue.get_style_context ();
+            color_button_blue_context.add_class ("color-button");
+            color_button_blue_context.add_class ("color-blue");
+
+            var color_button_violet = new Gtk.Button ();
+            color_button_violet.halign = Gtk.Align.CENTER;
+            color_button_violet.height_request = BUTTON_WIDTH;
+            color_button_violet.width_request = BUTTON_HEIGHT;
+            var color_button_violet_context = color_button_violet.get_style_context ();
+            color_button_violet_context.add_class ("color-button");
+            color_button_violet_context.add_class ("color-violet");
+
+            var color_button_slate = new Gtk.Button ();
+            color_button_slate.halign = Gtk.Align.CENTER;
+            color_button_slate.height_request = BUTTON_WIDTH;
+            color_button_slate.width_request = BUTTON_HEIGHT;
+            var color_button_slate_context = color_button_slate.get_style_context ();
+            color_button_slate_context.add_class ("color-button");
+            color_button_slate_context.add_class ("color-slate");
+
+            var color_button_remove = new Gtk.Button ();
+            color_button_remove.halign = Gtk.Align.CENTER;
+            color_button_remove.height_request = BUTTON_WIDTH;
+            color_button_remove.width_request = BUTTON_HEIGHT;
+            var color_button_remove_context = color_button_remove.get_style_context ();
+            color_button_remove_context.add_class ("color-button");
+
+            var colorbox = new Gtk.Box (Gtk.Orientation.HORIZONTAL, -1);
+            colorbox.pack_start (color_button_remove, true, true, 0);
+            colorbox.pack_start (color_button_red, true, true, 0);
+            colorbox.pack_start (color_button_orange, true, true, 0);
+            colorbox.pack_start (color_button_yellow, true, true, 0);
+            colorbox.pack_start (color_button_green, true, true, 0);
+            colorbox.pack_start (color_button_blue, true, true, 0);
+            colorbox.pack_start (color_button_violet, true, true, 0);
+            colorbox.pack_start (color_button_slate, true, true, 0);
+
             button_press_event.connect (button_pressed_cb);
```

Comment:
Connect each button to same handler and query signal source to determine which color pressed.

Reply:
How would that look like as an example?